\chapter{Inferring and modelling specification dynamics in retinal progenitors}
\label{chap:rysoutro}

\section{Specification as a function of non-proliferative chromatin dynamics}
The npat mutant \textit{rys} investigated in detail in \autoref{chap:rys} provides a useful denouement to the basic theme running through this thesis, the notion that lineage commitment or specification can be, and often is, dissociable from the proliferative behaviour of retinal progenitors, as has been suggested by Engerer et al. \cite{Engerer2017}. In \textit{rys}, we receive a graphic illustration of the sense in which this is so: RPCs undergo a final burst of proliferation before the death of the animal\footnote{Likely, on this view, arising from systemic failure of progenitors to specify appropriately.}, but never contribute appropriately to the neural retina. Indeed, many older \textit{rys} larvae display a proliferative spillage of abberant RPCs into the intraocular space between the lens and GCL; the progeny of these abberant RPCs literally cannot be integrated into the tissue.

That this dissociability has fundamental implications for the quality of our inferences about RPCs, and, therefore, our ability to predict and control their behaviour, can be gleaned from the basic problems encountered in each of the modelling studies. To begin with, the Stochastic Mitotic Mode Explanation models failed, in large part, because the underlying Simple Stochastic Models conflate proliferation and specification. This leads immediately to the introduction of model structure to explain specification outside of ``mitotic mode'', assuming a phased temporal structure to RPC behaviour, without attempting to explain it. It thus becomes plain that lineage specification and niche exit processes need to be modelled separately from mitotic processes. Next, proliferative data and morphological measurements largely fail to constrain the posterior distributions on the rate of niche exit in simple phased models, demonstrating that this parameter cannot be inferred with any certainty from overall population-level proliferative dynamics. This demonstrates that, in order to accurately model the behaviour of RPCs, data which provides accurate volumetric or temporal constraints on modelled specification or niche exit rate is of paramount importance. Finally, in \textit{rys}, we encounter the problem of explaining RPCs that proliferate but fail to specify; we explain this phenomenon in terms of the effects of mutant npat on chromatin organization and nucleosome sequence preferences. This last model directs us to the relevant level of biological organization for any rigorous explanation of RPC lineage outcomes: the progression of chromatin states that we take to underlay long-term changes in gene expression which are required for the generation of the specified protein complement. Our inferential approach, while providing good evidence for the chromatin dynamics we believe arise from the effects of abberant histone transcript regulation on the nucleosome pool, does not give us insight into how the appropriate progression of chromatin states might be usefully parameterised, beyond confirming that it is likely under the active control of \textit{trans}-acting factors.

While a large number of techniques for studying aspects of chromatin structure exist, few suggest model parameterisation for capturing an overall ``nuclear state''. That is, if we consider Waddington's idea of a cell's lineage state being a ball rolling down a canalized terrain, it remains unclear what sort of parameter space would allow us to infer where the ball is. Plainly, older ideas of cellular state space that rely exclusively on measures of protein expression are unable to address hypotheses about chromatin structure. One might, in the ``-omics'' enumerationist mode, propose a joint structure-sequence-expression space with thousands of variables drawn from the vast literature on chromatin organisation. Such a project is unlikely to succeed, even with the sophisticated Bayesian inference tools developed here, which can estimate the evidence for models with thousands of parameters across millions of observation. Practically speaking, it remains unclear what aspects of chromatin organisation are causally important in fate commitment, as little formal model comparison has taken place. Given this, prior distributions on the model parameters are unlikely to be well enough constrained to allow the convergence of ensembles on posterior distributions with reasonable compute budgets.

This raises the question of how we might structure models about specifiction-relevant nuclear state more simply. One rapidly developing line of thought, the school of biosemiotics championed by Marcello Barbieri and now referred to as code biology, may provide some insight. Code biology supposes that biological phenomena consist of physico-chemically determined phenomena (like the association of lipids into bilayers), and arbitrary coding relations between different ``worlds'' of biological macromolecules (including the genetic code and signalling codes).

transition from discussion of modelling postembryonic CMZ and using nuclear shape to infer identity for modelling purposes to discussion of the implication of the correlations of nuclear state, cellular identity, and cellular function

\section{Further directions}

\section{Frontiers of nuclear dynamics}
discuss recent experiments measuring chromatin conformation, pulsatile transcription etc, nuclear migration
\section{Integrating nuclear dynamics into models of CMZ- abstraction and biosemiotics}
abstraction necessary due to computational limits, biosemiotics allow for the calibration of this theoretically


\section{Summary}