\chapter{BioMotifInference: Independent component analysis motif inference by nested sampling for Julia}
\section{Introduction}
While many methods to detect overrepresented motifs in nucleotide sequences, only one is a rigorously validated system of statistical inference that allows us to calculate the evidence for these motif models, given genomic observations: nested sampling \cite{Skilling2006}. Computational biologists almost immediately benefitted from Skilling's original publication, with the release of the Java-coded nMICA by Down et al. in 2005 \cite{Down2005}. Unfortunately, this code base is no longer being maintained. It is, moreover, desireable to take advantage of modern languages and programming techniques to improve the maintainability and productivity of important bioinformatic code. We therefore re-implement Skilling's algorithm for inference of DNA motifs in Julia \cite{Bezanson2015} and characterise its function and performance.

\section{Implementation of the nested sampling algorithm}

\section{Recovery of spiked motifs}

\section{Recovery of promoter motifs}

\section{Performance}

\section{Discussion}